contracts/utils/Strings.sol

```diff
@@ -67,4 +67,12 @@ library Strings {
     function toHexString(address addr) internal pure returns (string memory) {
         return toHexString(uint256(uint160(addr)), _ADDRESS_LENGTH);
     }
+
+    /**
+     * @dev Returns true if the two strings are equal.
+     */
+    function compare(string memory a, string memory b) internal pure returns (bool) {
+        if (bytes(a).length != bytes(b).length) return false;
+        return keccak256(abi.encodePacked(a)) == keccak256(abi.encodePacked(b));
```

Comment:
Lets make use of lazy evaluation (also, the packing is not needed, and expensive)
```suggestion
        return bytes(a).length == bytes(b).length && keccak256(a) == keccak256(b);
```

Reply:
I'm curious how much the length check cost compared to what it saves. Is it really worth it ?

Reply:
Actually, the compiler gives error if I try with `keccak256(a) == keccak256(b)`, am I doing something wrong? Also, [here](https://github.com/0xCaso/test-strings-compare-function) I compared four different functions, and the below one seems the winner.
```suggestion
        return keccak256(bytes(a)) == keccak256(bytes(b));
```